### PR TITLE
Add publish button to Foorm editor

### PR DESCRIPTION
**Pull request by bencodeorg**
Adds a button to allow "publishing" of a survey via our Foorm editor. Currently this just flips a flag in the database and isn't tied to any functionality, but in the future will be used to identify surveys that are "production ready" and can be used in actual surveys of our users.

Publish option shown on "draft" surveys:

image

Warning when publishing button clicked:

image

Button disappears once survey is published:

image

## Testing story

Tested manually that I can publish surveys that are currently in draft state properly, and that the "publish" button does not appear for already published surveys.

## Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/code-studio/pd/foorm/editor/FoormSaveBar.jsx [hunk 1]
```diff
@@ -81,6 +81,21 @@ const publishedSaveWarning = (
   </div>
 );
 
+const aboutToPublishWarning = (
+  <div>
+    <span style={styles.warning}>Warning: </span>You are about to publish a new
+    survey. Once a survey is published, it may be put into active use.
```
Comment (molly-moen) [inline apps/src/code-studio/pd/foorm/editor/FoormSaveBar.jsx]: nit: do we want to add to the warning that you cannot go back to draft after you have published a survey?

File: apps/src/code-studio/pd/foorm/editor/FoormSaveBar.jsx [hunk 2]
```diff
@@ -118,26 +133,37 @@ class FoormSaveBar extends Component {
...
-      this.setState({showSaveConfirmation: true});
+      this.setState({
+        confirmationDialogBeingShownName: confirmationDialogNames.save
+      });
     } else if (this.props.formId === null || this.props.formId === undefined) {
       // if this is not an existing form, show new form save modal
       this.setState({showNewFormSave: true});
     } else {
-      this.save();
+      const saveUrl = `/foorm/forms/${this.props.formId}/update_questions`;
```
Comment (molly-moen) [inline apps/src/code-studio/pd/foorm/editor/FoormSaveBar.jsx]: should we put this url in a constant since it is used in multiple places?

**Review by molly-moen — APPROVED** [review]
Looks good! just a couple nits

**Comment by bencodeorg** [comment]
@molly-moen could you take a look at the last two commits? Didn't realize we had tests on this, so fixed existing ones and added a couple new ones.

File: apps/test/unit/code-studio/pd/foorm/FoormEditorTest.js [hunk 3]
```diff
@@ -115,6 +115,51 @@ describe('FoormEditor', () => {
...
+      JSON.stringify(sampleSaveData)
+    ]);
+
+    const saveBar = wrapper.find('FoormSaveBar');
+
+    const publishButton = saveBar.find('button').at(0);
+    expect(publishButton.contains('Publish')).to.be.true;
+    publishButton.simulate('click');
+
+    // check the the spinner is showing
```
Comment (molly-moen) [inline apps/test/unit/code-studio/pd/foorm/FoormEditorTest.js]: nit: remove redundant 'the'

**Review by molly-moen — APPROVED** [review]
Thanks for noticing the tests! Forgot to mention that

**Pull request merged**